definitions/npm/async_v1.x.x/flow_>=v0.28.x/async_v1.x.x.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+declare module 'async' {
+  declare type Collection<TItem> = Iterable<TItem> | { [key: string]: TItem };
+
+  declare type IterateeCallback<TError, TResult> = (err?: TError, result?: TResult) => void;
+  declare type Callback<TError, TResult> = (err?: TError, result?: TResult) => mixed;
+
+  // Collections
+  declare function each<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (item: TItem, callback: IterateeCallback<TError, void>) => mixed,
+    callback?: Callback<TError, void>
+  ): void;
+  declare function forEachOf<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (value: TItem, key: string | number, callback: IterateeCallback<TError, void>) => mixed,
+    callback?: Callback<TError, void>
+  ): void;
+  declare function map<TItem, TResult, TError>(
+    collection: Collection<TItem>,
+    iteratee: (value: TItem, callback: IterateeCallback<TError, TResult>) => mixed,
+    callback?: Callback<TError, Array<TResult>>
```

Comment:
I believe it's fixed locally, this definition is taking a really long time to get right and I haven't been able to focus on it.